composables/transaction/transactionBuy.ts

```diff
@@ -6,9 +6,35 @@ import {
 } from '@kodadot1/minimark/v2'
 
 import { getApiCall } from '@/utils/gallery/abstractCalls'
-import { isRoyaltyValid } from '@/utils/royalty'
 import { payRoyaltyTx, somePercentFromTX } from '@/utils/support'
 import type { ActionBuy } from './types'
+import { verifyRoyalty } from './utils'
+
+function payRoyaltyAssetHub(
+  legacy,
+  api,
+  price,
+  royalty,
+  collectionId,
+  tokenId,
+) {
+  const { isValid, normalizedRoyalty } = verifyRoyalty(royalty)
+
+  if (!isValid) {
+    return
+  }
+
+  return legacy
+    ? payRoyaltyTx(api, price, normalizedRoyalty)
+    : api.tx.nfts.payTips([
+        {
+          collection: collectionId,
+          item: tokenId,
+          receiver: normalizedRoyalty.address,
+          amount: royaltyFee(price, normalizedRoyalty.amount),
+        },
```

Comment:
move to utils?

Reply:
i don't see the value in it
you will have a function with 4 params plus API

why not just use `api.tx.nfts.payTips` directly?